### PR TITLE
Fix for #4053

### PR DESCRIPTION
**Pull request by GinoPane**
For more info about select2 placeholders: https://select2.org/placeholders.

Though its documentation says something against empty option for selects with `multiple` attribute, `allowClear` seems to behave correctly for tag lists, for example:
image

It adds an option to remove all tags by one click, which is pretty convenient, though it is a side effect 😄

### CODE REVIEW

**Comment by w20k** [comment]
@GinoPane could you recompile JS assets. `php artisan october:util compile js`

**Comment by GinoPane** [comment]
@w20k done 👍

File: modules/system/assets/ui/js/select.js [hunk 1]
```diff
@@ -113,6 +113,7 @@
             var placeholder = $element.data('placeholder')
             if (placeholder) {
                 extraOptions.placeholder = placeholder
+                extraOptions.allowClear = true;
```
Comment (w20k) [inline modules/system/assets/ui/js/select.js]: Please remove semicolon ;)

Comment (GinoPane) [reply modules/system/assets/ui/js/select.js]: Oops, done. Didn't pay attention that semicolons are not used there

**Comment by LukeTowers** [comment]
@w20k ping me once you're done testing. Refs: #4053

**Comment by w20k** [comment]
@LukeTowers I don't think this PR is related to #4053 .

PR is about clearButton. New option - allowClear
The issue is about enabling empty element in dropdown list aka placeholder. New option - allowEmpty 😉

**Comment by w20k** [comment]
I guess, works as expected, @LukeTowers.
Status - ✅
Screenshots:
screenshot 2019-01-26 at 22 52 27
screenshot 2019-01-26 at 22 52 48

**Comment by LukeTowers** [comment]
@GinoPane @w20k if that changes the existing behaviour by allowing dropdowns to be have a state of not having any option selected then that needs to be gated by a config property on the dropdown. Does it allow the selection to be cleared when the dropdown doesn't have a placeholder?

**Comment by LukeTowers** [comment]
@w20k so when a placeholder is not set, does that mean that the clear button isn't present?

**Comment by w20k** [comment]
@LukeTowers as for now, from js side - yes. 
Both options are set when placeholder is defined.

**Comment by LukeTowers** [comment]
@w20k Sweet. So is this ready to be merged now?

**Comment by w20k** [comment]
@LukeTowers Yep - ready!

**Pull request merged**